src/main/java/de/bonndan/nivio/input/GroupResolver.java

```diff
@@ -19,24 +28,52 @@ public GroupResolver(ProcessLog log) {
     }
 
     public void process(LandscapeDescription input, LandscapeImpl landscape) {
+
+        List<Function<String, Boolean>> specs = getSpecs(input.getConfig().getGroupBlacklist());
+
         input.getGroups().forEach((identifier, groupItem) -> {
             Group g = getGroup(identifier, groupItem);
 
-            log.info("Adding or updating group " + g.getIdentifier());
-            landscape.addGroup(g);
+            if (!isBlacklisted(g.getIdentifier(), specs)){
+                log.info("Adding or updating group " + g.getIdentifier());
+                landscape.addGroup(g);
+            } else {
+                log.info("Ignoring blacklisted group " + g.getIdentifier());
+            }
         });
 
-        input.getItemDescriptions().forEach(itemDescription -> {
+        ArrayList<ItemDescription> copy = new ArrayList<>(input.getItemDescriptions());
+        copy.forEach(item -> {
 
-            String group = itemDescription.getGroup();
-            if (StringUtils.isEmpty(itemDescription.getGroup())) {
+            String group = item.getGroup();
+            if (StringUtils.isEmpty(item.getGroup())) {
                 group = Group.COMMON;
             }
 
-            landscape.getGroups().computeIfAbsent(group, s -> getGroup(s, null));
+            if (!isBlacklisted(group, specs))
```

Comment:
On lines 37 - 42 you use curly braces for the if/else statement, but here not? Personally, I wouldn't omit them. But either way you should act consistenlty.

Reply:
has been fixed, thanks